syntax = "proto3";

package grpc;

message Empty {}

// I called it Note because message Message sounds complicated
message Note {
    string name = 1;
    string message = 2;
}

service ChatServer {
    // This bi-directional stream makes it possible to send and receive Notes between 2 persons
    rpc ChatStream (Empty) returns (stream Note);
    rpc SendNote (Note) returns (Empty);
    rpc PingPong(stream PingPong_t) returns (stream PingPong_t);
}

message PingPong_t {
    int64 num = 1;
}
